cockpit-lib-update

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Update COCKPIT_REPO_COMMIT to cockpit HEAD automatically, defaults to
+# Makefile as input optionally the full path can be provided. (For example
+# Anaconda uses ui/webui/Makefile.am).
+
+import os
+import re
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+sys.dont_write_bytecode = True
+
+import task
+from lib.constants import BASE_DIR
+
+GIT_URL_RE = r'COCKPIT_REPO_URL = (.*)'
+GIT_COMMIT_RE = r'COCKPIT_REPO_COMMIT = (.*)'
+
+
+def run(context, verbose=False, **kwargs):
+    cockpit_repo_url = 'https://github.com/cockpit-project/cockpit.git'
+    cockpit_repo_commit = 'HEAD'
+    makefile = context or 'Makefile'
+    makefile_path = os.path.join(BASE_DIR, makefile)
+
+    if not os.path.exists(makefile_path):
+        raise FileNotFoundError(f"Makefile '{makefile_path}' does not exists")
```

Comment:
The `open()` below will do pretty much this, so this is redundant (and the "exist*s*" is also ungrammatical)